File: 2021/UICPC DIV1/UICPC Round #05 (Div 1)/solutions/Problem G - Inflation/inflation.cpp
#include <bits/stdc++.h>
using namespace std;

// Inflation

int n, a[200003];

int main(){
    cin >> n;
    for(int i = 0; i < n; i++)
        cin >> a[i];
    sort(a, a + n);
    double ans = 1;
    for(int i = 0; i < n; i++){
        if(a[i] > i + 1){
            cout << "impossible" << endl;
            return 0;
        }
        ans = min(ans, double(a[i]) / (i + 1));
    }
    cout << fixed << setprecision(10) << ans << endl;
}

File: 2021/UICPC DIV1/UICPC Round #05 (Div 1)/solutions/Problem G - Inflation/inflation.py
n = int(input())
li = list(map(int, input().split()))
li.sort()
mi = 2
for i in range(n):
    if li[i] > i+1:
        print('impossible')
        exit(0)
    mi = min(mi, li[i]/(i+1))
print(mi)